diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54e6a9f..47243f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.12)
 
 project(picotool)
 
+find_package(MbedTLS REQUIRED)
+
 if (NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release)
 endif()
@@ -102,7 +104,6 @@ add_embedded_data_project(enc_bootloader
 
 set(ENC_BOOTLOADER_ELF ${CMAKE_BINARY_DIR}/enc_bootloader/enc_bootloader.elf)
 
-if (TARGET mbedtls)
     add_embedded_data_project(enc_bootloader_mbedtls
             PREFIX enc_bootloader_mbedtls
             SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/enc_bootloader
@@ -116,7 +117,6 @@ if (TARGET mbedtls)
             )
 
     set(ENC_BOOTLOADER_MBEDTLS_ELF ${CMAKE_BINARY_DIR}/enc_bootloader_mbedtls/enc_bootloader.elf)
-endif()
 
 if (NOT PICOTOOL_NO_LIBUSB)
     # compile xip_ram_perms.elf
@@ -207,14 +207,14 @@ if (NOT PICOTOOL_NO_LIBUSB)
     endif()
 endif()
 
-if (TARGET mbedtls)
+#if (TARGET mbedtls)
     add_custom_target(embedded_data_no_libusb DEPENDS
             ${CMAKE_CURRENT_BINARY_DIR}/enc_bootloader_elf.h
             ${CMAKE_CURRENT_BINARY_DIR}/enc_bootloader_mbedtls_elf.h)
-else()
-    add_custom_target(embedded_data_no_libusb DEPENDS
-            ${CMAKE_CURRENT_BINARY_DIR}/enc_bootloader_elf.h)
-endif()
+#else()
+#    add_custom_target(embedded_data_no_libusb DEPENDS
+#            ${CMAKE_CURRENT_BINARY_DIR}/enc_bootloader_elf.h)
+#endif()
 
 add_custom_target(embedded_data DEPENDS
         ${CMAKE_CURRENT_BINARY_DIR}/xip_ram_perms_elf.h
@@ -319,14 +319,16 @@ target_link_libraries(picotool
         elf2uf2
         errors
         nlohmann_json
-        whereami)
+        whereami
+        MbedTLS::mbedtls)
 
-if (NOT TARGET mbedtls)
-    message("mbedtls not found - no signing/hashing support will be built")
-    target_compile_definitions(picotool PRIVATE HAS_MBEDTLS=0)
-else()
+#if (NOT TARGET mbedtls)
+#    message("mbedtls not found - no signing/hashing support will be built")
+#    target_compile_definitions(picotool PRIVATE HAS_MBEDTLS=0)
+#else()
     target_compile_definitions(picotool PRIVATE HAS_MBEDTLS=1)
-endif()
+#endif()
+target_link_libraries(picotool MbedTLS::mbedtls MbedTLS::mbedcrypto)
 
 if (NOT LIBUSB_FOUND)
     if (PICOTOOL_NO_LIBUSB)
@@ -390,14 +392,14 @@ install(FILES
     DESTINATION ${INSTALL_DATADIR}
 )
 
-if (TARGET mbedtls)
+#if (TARGET mbedtls)
     #Install enc_bootloader_mbedtls.elf
     install(FILES
         ${ENC_BOOTLOADER_MBEDTLS_ELF}
         DESTINATION ${INSTALL_DATADIR}
         RENAME enc_bootloader_mbedtls.elf
     )
-endif()
+#endif()
 
 if (NOT PICOTOOL_NO_LIBUSB)
     if (NOT PICOTOOL_CODE_OTP)
